datumaro/components/hl_ops.py

```diff
@@ -0,0 +1,192 @@
+    media and annotations.
+
+    Args:
+        dataset (IDataset) - The dataset to be transformed
+        model (Launcher or Modeltransform) - The model to be
+            applied to the dataset.
+        batch_size (int) - The number of dataset items, processed
+            simultaneously by the model
+        **kwargs - Optional parameters for the model
+
+    Returns: a lazy-computible wrapper around the input dataset
+    """
+
+    if isinstance(model, Launcher):
+        return transform(dataset, ModelTransform, launcher=model,
+            batch_size=batch_size, **kwargs)
+    elif isinstance(model, ModelTransform):
+        return transform(model, batch_size=batch_size, **kwargs)
+    else:
+        raise TypeError('Unexpected model argument type: %s' % type(model))
+
+# other operations
+
+@scoped
+def export(dataset: IDataset, path: str, format: Union[str, Converter], *,
+        env: Optional[Environment] = None, **kwargs) -> None:
+    """
+    Saves the input dataset in some format.
+
+    Args:
+        dataset (IDataset) - The dataset to be saved
+        format (str or Converter) - The desired output format for the dataset.
+            If a string is passed, it is treated as a plugin name,
+            which is searched for in the environment set by the 'env' argument
+        env (Environment) - A plugin collection. If not set, the default one
+            is used
+        **kwargs - Optional parameters for the export format
+    """
+
+    if isinstance(format, str):
+        if env is None:
+            env = Environment()
+        converter = env.converters[format]
+    else:
+        converter = format
+
+    path = osp.abspath(path)
+    if not osp.exists(path):
+        on_error_do(shutil.rmtree, path, ignore_errors=True)
+    os.makedirs(path, exist_ok=True)
+
+    converter.convert(dataset, save_dir=path, **kwargs)
+
+def detect(path: str, *,
+        env: Optional[Environment] = None, depth: int = 1) -> str:
+    """
+    Attempts to detect dataset format of a given directory.
+
+    Args:
+        path (str) - The directory to check
+        depth (int) - The maximum depth for recursive search
+        env (Environment) - A plugin collection. If not set, the default one
+            is used
+    """
+
+    return Dataset.detect(path, env=env, depth=depth)
+
+def validate(dataset: IDataset, task: Union[str, TaskType], *,
+        env: Optional[Environment] = None, **kwargs) -> Dict:
+    """
+    Checks dataset annotations for correctness relatively to a task type.
+
+    Args:
+        dataset (IDataset) - The dataset to check
+        task (str or TaskType) - Target task type - classification, detection
+            etc.
+        env (Environment) - A plugin collection. If not set, the default one
+            is used
+        **kwargs - Optional arguments for the validator
+
+    Returns: a dictionary with validation results
+    """
+
+    task = parse_str_enum_value(task, TaskType).name.lower()
```

Comment:
Why is `lower` needed here? The members of `TaskType` are already lowercase.

Reply:
Actually, not to think about it. Probably, it's better to be checked in tests. Added a test for this.